jib-core/src/test/java/com/google/cloud/tools/jib/configuration/ContainerConfigurationTest.java

```diff
@@ -0,0 +1,76 @@
+package com.google.cloud.tools.jib.configuration;
+
+import static org.junit.Assert.*;
+
+import com.google.cloud.tools.jib.configuration.Port.Protocol;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ContainerConfigurationTest {
+
+  @Test
+  public void testBuilder_nullValues() {
+    // Java arguments element should not be null.
+    try {
+      ContainerConfiguration.builder().setProgramArguments(Arrays.asList("first", null));
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // Entrypoint element should not be null.
+    try {
+      ContainerConfiguration.builder().setEntrypoint(Arrays.asList("first", null));
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // Exposed ports element should not be null.
+    try {
+      ContainerConfiguration.builder()
+          .setExposedPorts(Arrays.asList(new Port(1000, Protocol.TCP), null));
```

Comment:
This is still not narrow enough. For example, suppose Port was modified to throw IllegalArgumentException when given 1000. The test will still pass if we introduce a bug into .setExposedPorts() that won't throw the exception.

Reply:
Ah right. Will update